packages/web3-eth-abi/src/api/events_api.ts

```diff
@@ -0,0 +1,23 @@
+import { AbiError } from 'web3-common';
+import { sha3Raw } from 'web3-utils';
+import { JsonAbiEventFragment } from '../types';
+import { jsonInterfaceMethodToString, isAbiEventFragment } from '../utils';
+
+/**
+ * Encodes the function name to its ABI representation, which are the first 4 bytes of the sha3 of the function name including  types.
+ */
+export const encodeEventSignature = (functionName: string | JsonAbiEventFragment): string => {
+	if (typeof functionName !== 'string' && !isAbiEventFragment(functionName)) {
+		throw new AbiError('Invalid parameter value in encodeEventSignature');
+	}
+
+	let name: string;
+
+	if (typeof functionName === 'function' || (typeof functionName === 'object' && functionName)) {
```

Comment:
Just curious when `functionName`'s type is `function`. The [JsonAbiEventFragment](https://github.com/ChainSafe/web3.js/blob/7b63261513377283c02304885c2e37eff082dab0/packages/web3-eth-abi/src/types.ts#L45) is only ever an object:


```typescript
export interface JsonAbiEventFragment extends JsonAbiBaseFragment {
	type: 'event';
	anonymous?: boolean;
	outputs?: Array<JsonAbiParameter>;
}
```

Reply:
Additionally, `jsonInterfaceMethodToString` doesn't account for `functionName` being of type `function`

Reply:
[1.x implementation](https://github.com/ChainSafe/web3.js/blob/1.x/packages/web3-eth-abi/src/index.js#L69) checks for `typeof functionName === 'function'` as well (probably a carry over, but that means our Types don't make sense for the method)

Reply:
I got the point, but this code been refactored in coming PR. So we can proceed with this code as it is.